sky/packages/sky/lib/widgets/framework.dart

```diff
@@ -121,15 +145,31 @@ abstract class GlobalKey extends Key {
     assert(!Widget._notifyingMountStatus);
     assert(_debugDuplicates.isEmpty);
     _notifyingListeners = true;
-    for (GlobalKey key in _removedKeys) {
-      if (!_registry.containsKey(key) && _removalListeners.containsKey(key)) {
-        for (GlobalKeyRemovalListener listener in _removalListeners[key])
-          listener(key);
-        _removalListeners.remove(key);
+    try {
```

Comment:
Maybe wrap the work here around if blocks that first verify that there's any work to do?
Currently in the no-op case there's a lot of busywork (creating a new map, copying the listeners, exception handling...).